lib/kafka/sasl/awsmskiam.rb

```diff
@@ -90,10 +135,10 @@ def canonical_query_string(time_now:)
         params = {
           "Action" => "kafka-cluster:Connect",
           "X-Amz-Algorithm" => "AWS4-HMAC-SHA256",
-          "X-Amz-Credential" => @access_key_id + "/" + time_now.strftime("%Y%m%d") + "/" + @aws_region + "/kafka-cluster/aws4_request",
+          "X-Amz-Credential" => @aws_msk_iam_credentials.get_access_key_id + "/" + time_now.strftime("%Y%m%d") + "/" + @aws_region + "/kafka-cluster/aws4_request",
           "X-Amz-Date" => time_now.strftime("%Y%m%dT%H%M%SZ"),
           "X-Amz-Expires" => "900",
-          "X-Amz-Security-Token" => @session_token,
+          "X-Amz-Security-Token" => @aws_msk_iam_credentials.get_session_token,
```

Comment:
does this get messed up if the session token isn't passed and defaults to `nil`?

Reply:
should still work.. `.delete_if { |_, v| v.nil? }` will remove `"X-Amz-Security-Token" => nil`